pkg/broker/broker.go

```diff
@@ -789,21 +769,47 @@ func (a AnsibleBroker) Unbind(
 		return nil, errors.New(errMsg)
 	}
 
+	params := make(apb.Parameters)
+	provExtCreds, err := a.dao.GetExtractedCredentials(instanceUUID.String())
+	if err != nil {
+		return nil, err
+	}
+	bindExtCreds, err := a.dao.GetExtractedCredentials(bindingUUID.String())
+	if err != nil && !client.IsKeyNotFound(err) {
+		return nil, err
+	}
+	// Add the credentials to the parameters so that an APB can choose what
+	// it would like to do.
+	params[provisionCredentialsKey] = provExtCreds.Credentials
+	if bindExtCreds != nil {
+		params[bindCredentialsKey] = bindExtCreds.Credentials
+	}
 	serviceInstance, err := a.getServiceInstance(instanceUUID)
 	if err != nil {
 		a.log.Debugf("Service instance with id %s does not exist", instanceUUID.String())
 	}
+	if serviceInstance.Parameters != nil {
+		params["provision_params"] = *serviceInstance.Parameters
+	}
+	// only launch apb if we are always launching the APB.
+	if a.brokerConfig.LaunchApbOnBind {
+		err = apb.Unbind(serviceInstance, a.clusterConfig, a.log, &params)
+		if err != nil {
+			return nil, err
+		}
+	}
 
-	err = apb.Unbind(serviceInstance, a.clusterConfig, a.log)
-	if err != nil {
-		return nil, err
+	if bindExtCreds != nil {
+		err = a.dao.DeleteExtractedCredentials(bindingUUID.String())
+		if err != nil {
```

Comment:
I see 4 cases on [777](https://github.com/openshift/ansible-service-broker/pull/302/files/63fb106568574aae52036b0b4b219bdeee23bbf3#diff-55aca6fb74ac953881309c295ee9441eR777) we will attempt to get the credentials.
1. etcd/dao error that is something other than creds could not be found
  - we return the error and error out to service catalog which will re-try.
2. etcd/dao error that is creds not found
  - we don't need to attempt to delete anything that is not there
3. etcd/dao finds creds and delete succeeds 
  - happy path
4. etcd/dao finds creds and delete creds fails.
  - we will error out which will bubble up to service catalog which will re-try.


Can you think of another path?